pkg/proxy/endpointslicecache_test.go

```diff
@@ -365,9 +422,20 @@ func generateEndpointSliceWithOffset(serviceName, namespace string, sliceNum, of
 	}
 
 	for i := 1; i <= numEndpoints; i++ {
+		readyCondition := i%unreadyMod != 0
+		terminatingCondition := i%terminatingMod == 0
+
+		ready := utilpointer.BoolPtr(readyCondition && !terminatingCondition)
+		serving := utilpointer.BoolPtr(readyCondition)
+		terminating := utilpointer.BoolPtr(terminatingCondition)
+
 		endpoint := discovery.Endpoint{
-			Addresses:  []string{fmt.Sprintf("10.0.%d.%d", offset, i)},
-			Conditions: discovery.EndpointConditions{Ready: utilpointer.BoolPtr(i%unreadyMod != 0)},
+			Addresses: []string{fmt.Sprintf("10.0.%d.%d", offset, i)},
```

Comment:
This test should probably panic if you ask for more than 255 endpoints (not a new problem, but might as well while we're in here)